cluster/cluster.go

```diff
@@ -0,0 +1,83 @@
+package cluster
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/rke/hosts"
+	"github.com/rancher/rke/k8s"
+	"github.com/rancher/rke/services"
+	yaml "gopkg.in/yaml.v2"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+type Cluster struct {
+	Services services.Services `yaml:"services"`
```

Comment:
Lets think about defining all core structs in a separate file. This way it will be easier to refer to them. Also I see that you are calling divideHosts logic from many places. A config file passed by user, should be parsed and transformed to cluster data structure just once. If you think that there are some extra data structures are needed by your code - etcdHosts/controlPlaneHosts/etc - introduce them